assets/training/finetune_acft_hf_nlp/components/finetune/text_generation/spec.yaml

```diff
@@ -0,0 +1,336 @@
+
+  adam_epsilon:
+    type: number
+    default: 1e-8
+    optional: true
+    description: The epsilon hyperparameter for the AdamW optimizer
+
+  gradient_accumulation_steps:
+    type: integer
+    default: 1
+    optional: true
+    description: Number of updates steps to accumulate the gradients for, before performing a backward/update pass
+
+  eval_accumulation_steps:
+    type: integer
+    default: 1
+    optional: true
+    description: Number of predictions steps to accumulate before moving the tensors to the CPU
+
+  lr_scheduler_type:
+    type: string
+    default: linear
+    optional: true
+    enum:
+      - linear
+      - cosine
+      - cosine_with_restarts
+      - polynomial
+      - constant
+      - constant_with_warmup
+    description: The scheduler type to use
+
+  precision:
+    type: string
+    enum:
+      - "32"
+      - "16"
+    default: "32"
+    optional: true
+    description: Apply mixed precision training. This can reduce memory footprint by performing operations in half-precision.
+
+  seed:
+    type: integer
+    default: 42
+    optional: true
+    description: Random seed that will be set at the beginning of training
+
+  enable_full_determinism:
+    type: string
+    enum:
+      - "true"
+      - "false"
+    default: "false"
+    optional: true
+    description: Ensure reproducible behavior during distributed training
+
+  dataloader_num_workers:
+    type: integer
+    default: 0
+    optional: true
+    description: Number of subprocesses to use for data loading. 0 means that the data will be loaded in the main process.
+
+  ignore_mismatched_sizes:
+    type: string
+    enum:
+      - "true"
+      - "false"
+    default: "true"
+    optional: true
+    description: Whether or not to raise an error if some of the weights from the checkpoint do not have the same size as the weights of the model
+
+  max_grad_norm:
+    type: number
+    default: 1.0
+    optional: true
+    description: "Maximum gradient norm (for gradient clipping)"
+
+  evaluation_strategy:
+    type: string
+    default: epoch
+    optional: true
+    enum:
+      - epoch
+      - steps
+    description: The evaluation strategy to adopt during training
+
+  evaluation_steps_interval:
+    type: number
+    default: 0.0
+    optional: true
+    description: The evaluation steps in fraction of an epoch steps to adopt during training. Overwrites evaluation_steps if not 0.
+
+  eval_steps:
+    type: integer
+    default: 500
+    optional: true
+    description: Number of update steps between two evals if evaluation_strategy='steps'
+
+  logging_strategy:
+    type: string
+    default: epoch
+    optional: true
+    enum:
+      - epoch
+      - steps
+    description: The logging strategy to adopt during training.
+
+  logging_steps:
+    type: integer
+    default: 500
+    optional: true
+    description: Number of update steps between two logs if logging_strategy='steps'
+
+  metric_for_best_model:
```

Comment:
Is _bleu_ valid as part of _metric_for_best_model_?

Reply:
No its not, removed it